frontend/helpers/generateFakeSurplusAuction.ts

```diff
@@ -68,7 +68,7 @@ export const generateFakeSurplusAuctionTransaction = function (
     const surplusAuction = generateFakeSurplusAuction(state);
 
     // if auction does not have any bids yet or is finished return without market data
-    if (!surplusAuction.bidAmountMKR || !surplusAuction.receiveAmountDAI) {
+    if (!surplusAuction.bidAmountMKR || !surplusAuction.receiveAmountDAI || surplusAuction.state === 'just-started') {
```

Comment:
I would generally do that, however this is more a typescript issue. Typescript does not know that every Auction aside from just-started will have these values and therefore throws errors. 

As this is only a generation file for mock data, I did not put too much effort into making sure that typescript would identify it here in this case. Thats why I only have the values.

Reply:
Very good point, since it can actually be (and I would argue _should be_) expressed via types. You can define separate `SurplusAuctionStarted` type and make `SurplusAuction` to be a union between those. See https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions for example.

Reply:
I ran into issues using this approach when writing the generation files. I sadly cannot find the commit where this error occurred, but I will outline it here:

types:
```ts
interface BaseAuctionData {
   id: number,
   network: string,
}

interface CollectedAuction extends BaseAuctionData {
   state: 'collected'
}

interface ActiveAuction extends BaseAuctionData {
   state: 'just-started', 'have-bids'
}

interface Auction = ActiveAuction || CollectedAuction

interface AuctionTransaction extends Auction {
   unitPrice: number,
   marketUnitPrice: number,
}
```

I then had one function that generated a `Auction`, randomly determining the state. When I then wanted to generate a AuctionTransaction, I would begin by creating a Auction (with a set state of `have-bids`, so ensure no collected Auction was generated).
Then I would take the data from the generated Auction to compute the values needed for the AuctionTransaction.

```ts
function generateFakeAuction(state?: string): Auction {
   ...generate mock data for auction based on provided state and if none is provided a random state.
}

function generateFakeAuctionTrasnaction(): AuctionTransaction {
   auction = generateFakeAuction('have-bids');

  const unitPrice = auction.totalAmountDAI.dividedBy(auction.bidAmountMKR)
  ...etc.
}
```

I would then get a typescript error thrown at the line:
```ts
const unitPrice = auction.totalAmountDAI.dividedBy(auction.bidAmountMKR)
```

due to `totalAmountDAI` and `bidAmountMKR` not existing on a completed Auction. As far as I could see I had two choices.

- Dont create new types for separate Auction States and have it all be handled via just Auction.
- Create seperate generation functions for the Auctions (such as `generateFakeActiveAuction` and `generateFakeCompletedAuction`)

Ultimately, I decided to use option two, as I also feared, I would run into more similar type based issues on a component or store level (I don't know if this fear was justified, but it did lead me to make my decision). 

I hope this text helped clear up the issue I ran into and my thought process 🤗